src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
This is there for a reason.  @vmg commented on his pull request: "Every single node is now also aware of its allocator, which means that we can choose an allocator when creating new nodes and check that you cannot insert a node from a specific allocator into a document tree created by another allocator."  @vmg and @nwellnhof can perhaps comment further on why this is needed.  It shouldn't prevent adding nodes created using `cmark_node_new` to the AST; you just need to make sure that the new node is created using the same memory allocator as the original document, right?